### ISSUE TITLE
train command for CAMERA dataset

### ISSUE THREAD

**Issue by skarapanahalli** [zubair-irshad/CenterSnap#24]
hi,
 Excellent work Irshad & team !. 
I am trying to follow the instructions on github to train. I have setup a windows environment with GPU support and trying to run jupyter notebook commands. (have downloaded both code & datasets)
However, when I try to run the below code
%run net_train.py "@configs/net_config.txt"
It is resulting in div by zero error

_ZeroDivisionError                         Traceback (most recent call last)
D:\aiexpts\6dof\gcasp\CenterSnap\net_train.py in <module>
     59   steps = hparams.max_steps
     60   steps_per_epoch = samples_per_epoch // samples_per_step
---> 61   epochs = int(np.ceil(steps / steps_per_epoch))
     62   actual_steps = epochs * steps_per_epoch
     63   print('Samples per epoch', samples_per_epoch)_

When I tried to debug, I found that the simnet code is actually looking for pickle files (self.dataset_path.glob('*.pickle.zstd')). But CAMERA & Real datasets are not having these files.

am I missing something ??

**Comment by zubair-irshad** [comment]
Hello @skarapanahalli, Thanks for finding our work interesting. For jupyter notebooks, you don't need downloaded data or pickle files but for training our model, you need to download or create data locally as we mention [here](https://github.com/zubair-irshad/CenterSnap#-dataset).

**Comment by zubair-irshad** [comment]
@skarapanahalli We haven't seen this error before but looks like a conda environment package mismatch error maybe? since it does not point anywhere in the code itself so it is hard to debug which line triggered this error. Can you please share the full error trace in a more readable (code format) please.

**Comment by skarapanahalli** [comment]
@zubair-irshad , the error is due to wandb issue. I am not able to perform wandb.init(), it is complaning of some proxy issue even though I have everything set correctly.
I can proceed further if I comment the logger line below in net_train.py. But it fails later, perhaps related to logs.

    trainer = pl.Trainer(
        max_nb_epochs=epochs,
        early_stop_callback=None,
        gpus=[_GPU_TO_USE],
        checkpoint_callback=model_checkpoint,
        val_check_interval=1.0,
        **logger=wandb_logger,**
        fast_dev_run=True,
        default_save_path=hparams.output,
        use_amp=False,
        print_nan_grads=True,
        
Is it possible to run the code without wandb logger ?

**Comment by zubair-irshad** [comment]
Did you do `wandb.login()` in the command line to set up wandb? Yes it is quite easy to change it to any other logger, for instance tensorboard. You can follow PyTorch Lightning's online documentation to read up on that. Hope it resolves your issue.

**Comment by skarapanahalli** [comment]
Yes I did wandb.login() and it is resulting in the same error. After much googling, I found that I need to update wandb, however when I try to upgrade it is having other dependencies and not easy to resolve. 
the issue is not code, but the handling of proxies by urllib3 as described here
https://stackoverflow.com/questions/66642705/why-requests-raise-this-exception-check-hostname-requires-server-hostname

**Comment by zubair-irshad** [comment]
I see. In that case, it might be worth switching to a tensorboard logger which is very easy to swap in PyTorch lightning with just a single line of code and a few import lines: https://www.pytorchlightning.ai/blog/tensorboard-with-pytorch-lightning

Please note that you might have to change a few image logging functions inside our code as well to make it work with tensorboard logging but hopefully the link above would help you with this.

**Comment by skarapanahalli** [comment]
Thanks @zubair-irshad, finally I am able to execute the code (training part). I switched to tensorboard, but TB visualizations won't work due to compatibility issues with older packages. I have now found a work around to copy the tb_logs to my desktop and upload it to tensorboard dev (online) and view it there !
When you get time I request you to update the code to run on latest packages, or may be create a docker image which we can use directly.  
Nevertheless, appreciate for your quick responses !
